wordpress-mobile/WordPress-iOS
# Fix preview's done button

Fixes #6705

Our check for modal presentation was returning true for all view controllers within a nav controller which had been presented modally. Now it will only be true for the first such view controller.

To test:
_For preview while editing_
1. Open the posts list on a site.
2. Tap "Edit" or tap the post card to open a post in the editor.
3. Tap the "..." menu in the top right and preview the post.
4. Ensure that no "Done" button appears

_For post-post preview_
1. Publish a post
2. Tap "View Post" from the post-post screen
3. Ensure that the "Done" button appears and is functional

Needs review: @jleandroperez 
Can you also check that the code I changed in `SettingsTextViewController.m` still works? I believe this was a problem somewhere within people management, but I couldn't trigger it.
